packages/api/src/server/environment/service/factory.ts

```diff
@@ -36,40 +36,63 @@ export class EnvironmentFactory {
   }
 
   /* @internal
-   * Environment initialization and the config creation
+   * Environment initialization, rebuild and the config creation
    * Creation of environment in the cluster
    */
-  async initializeEnvironment(propertyRequest: Property, environmentRequest: Environment) {
-    const initializeEnvironment = new CreateApplicationDto();
-    initializeEnvironment.name = 'envInit';
-    initializeEnvironment.ref = '0.0.0';
-    initializeEnvironment.path = '.';
+  async configureEnvironment(
+    buildEnvironment: CreateApplicationDto,
+    fileOriginalName: string,
+    zipPath: string,
+    propertyRequest: Property,
+    environmentRequest: Environment,
+    rebuild: boolean = false
+  ) {
+    try {
+      await this.applicationService.deployApplication(
+        buildEnvironment,
+        fileOriginalName,
```

Comment:
```suggestion
        orginalFIleName,
```

Reply:
Hi @riginoommen , I've changed it to `originalFileName`, please check